src/thunderbots/software/ai/hl/stp/evaluation/indirect_chip.h

```diff
@@ -0,0 +1,147 @@
+     *
+     * Given the vector of triangles without enemy robots, determines if the triangles are
+     * empty and if the largest triangle is within reach. If not within reach, scale the
+     * target point with the maximum chip power. The target point is where ball will land
+     * according to chipping calibration, as well as where the chaser will meet the ball
+     * at.
+     *
+     * @param triangles A vector of triangles with no enemy robots and are within best
+     * chip target area
+     * @param ball_position Position of the ball
+     *
+     * @return Point to chip and chase at; If null, the target triangles were empty
+     */
+    std::optional<Point> findTargetPointForIndirectChipAndChase(
+        const std::vector<Triangle>& triangles, Point ball_position);
+
+    /**
+     * Returns a vector of all possible triangles between enemy players and rectangular
+     * chip area.
+     *
+     * Creates a vector of triangles that picks all permutations of points from the
+     * list of all non-goalie enemy players as well as the four points returned by
+     * get_chip_area_target_corners.
+     *
+     * @param world The world in which we want to find the target point
+     * @param enemy_players Vector of enemy robots' positions
+     *
+     * @return Vector of triangles
+     */
+    std::vector<Triangle> getAllTrianglesBetweenEnemyPlayers(
+        const World& world, std::vector<Point> enemy_players);
+
+    /**
+     * Returns a vector of triangles that only contains triangles without enemy robots.
+     *
+     * Given a vector of triangles, shrink each triangle slightly (to avoid counting the
+     * robot's width within the triangle), filters out the triangles that consist of enemy
+     * robots and returns a new vector that only contains triangles without enemy robots.
+     *
+     * @param triangles Vector of triangles
+     * @param enemy_players Vector of enemy robots' positions
+     *
+     * @return Vector of triangles with no enemy robots
+     */
+    std::vector<Triangle> findOpenTriangles(std::vector<Triangle> triangles,
+                                            std::vector<Point> enemy_players);
+
+    /**
+     * Returns the center point the given triangle.
+     *
+     * @param triangle
+     *
+     * @return Centre point of triangle
+     */
+    Point getTriangleCenter(Triangle triangle);
+
+    /**
+     * Returns the area of the given triangle.
+     *
+     * @param triangle
+     *
+     * @return Area of triangle
+     */
+    double getTriangleArea(Triangle triangle);
+
+    /**
+     * Remove all Triangles in a given list whose centers do not fall
+     * within the rectangle returned by get_chip_target_area.
+     *
+     * @param world The world in which we want to find the target point
+     * @param triangles
+     *
+     * @return Valid triangles
+     */
+    std::vector<Triangle> remove_outofbounds_triangles(const World& world,
```

Comment:
we should rename to `removeOutOfBoundsTriangles`

Reply:
Edited as suggested